code/game/gamemodes/dynamic/dynamic_storytellers.dm

```diff
@@ -0,0 +1,149 @@
+		if (rule.acceptable(mode.current_players[CURRENT_LIVING_PLAYERS].len, mode.threat_level) && mode.threat >= rule.cost)
+			// Classic secret : only autotraitor/minor roles
+			if (GLOB.dynamic_classic_secret && !((rule.flags & TRAITOR_RULESET) || (rule.flags & MINOR_RULESET)))
+				continue
+			rule.trim_candidates()
+			if (rule.ready())
+				var/property_weight = 0
+				for(var/property in property_weights)
+					if(property in rule.property_weights)
+						property_weight += rule.property_weights[property] * property_weights[property]
+				drafted_rules[rule] = rule.get_weight() + property_weight
+		else if(mode.threat < rule.cost)
+			SSblackbox.record_feedback("tally","dynamic",1,"Times rulesets rejected due to not enough threat to spend")
+	return drafted_rules
+
+/datum/dynamic_storyteller/proc/latejoin_draft(mob/living/carbon/human/newPlayer)
+	var/list/drafted_rules = list()
+	for (var/datum/dynamic_ruleset/latejoin/rule in mode.latejoin_rules)
+		if (rule.acceptable(mode.current_players[CURRENT_LIVING_PLAYERS].len, mode.threat_level) && mode.threat >= rule.cost)
+			// Classic secret : only autotraitor/minor roles
+			if (GLOB.dynamic_classic_secret && !((rule.flags & TRAITOR_RULESET) || (rule.flags & MINOR_RULESET)))
+				continue
+			// No stacking : only one round-ender, unless threat level > stacking_limit.
+			if (mode.threat_level > GLOB.dynamic_stacking_limit && GLOB.dynamic_no_stacking)
+				if(rule.flags & HIGHLANDER_RULESET && mode.highlander_executed)
+					continue
+
+			rule.candidates = list(newPlayer)
+			rule.trim_candidates()
+			if (rule.ready())
+				var/property_weight = 0
+				for(var/property in property_weights)
+					if(property in rule.property_weights)
+						property_weight += rule.property_weights[property] * property_weights[property]
+				drafted_rules[rule] = rule.get_weight() + property_weight
+		else if(mode.threat < rule.cost)
+			SSblackbox.record_feedback("tally","dynamic",1,"Times rulesets rejected due to not enough threat to spend")
+	return drafted_rules
+
+/datum/dynamic_storyteller/proc/event_draft()
+	var/list/drafted_rules = list()
+	for(var/datum/dynamic_ruleset/event/rule in mode.events)
+		if(rule.acceptable(mode.current_players[CURRENT_LIVING_PLAYERS].len, mode.threat_level) && mode.threat >= rule.cost)
+			if(rule.ready())
+				var/property_weight = 0
+				for(var/property in property_weights)
+					if(property in rule.property_weights)
+						property_weight += rule.property_weights[property] * property_weights[property]
+				drafted_rules[rule] = rule.get_weight() + property_weight
+		else if(mode.threat < rule.cost)
+			SSblackbox.record_feedback("tally","dynamic",1,"Times rulesets rejected due to not enough threat to spend")
+	return drafted_rules
+
+
+/datum/dynamic_storyteller/cowabunga
+	name = "Chaotic"
+	curve_centre = 10
+	desc = "Chaos: high. Variation: high. Likely antags: clock cult, revs, wizard."
```

Comment:
In a future, these sort of descriptions will have to be actually classified on New() based on factors such as their property weights and accessible rulesets for easier maintenance, not gonna lie.

Reply:
not entirely sure i want them permanently at all